### DISCUSSION TITLE
capacitor version for deno-ts dev ?

### DISCUSSION THREAD

**Discussion by elycheikhsmail** [ionic-team/capacitor#4629] [General]
is possible to run capacitor cli equivalent using deno as js/ts runtime instad of nodejs ? is capacitor planning this ?

**Comment by mlynch** [comment]
Well Capacitor doesn't run on Node (except on Electron I suppose), it uses the web engine available on each platform. For example, that's the WebKit engine in `WKWebView` and the Chrome engine on Android. Neither of those are Node contexts, so the idea of swapping it out with a different JS engine, while possible for some non-web operations (such as those that would be done with `JavaScriptCore`), isn't possible for the rest of the app.

**Reply by jcesarmobile** [reply]
mlynch, the question is about @capacitor/cli.

There are no plans to switch to deno as runtime at the moment.

**Reply by elycheikhsmail** [reply]
> mlynch, the question is about @capacitor/cli.
> 
> There are no plans to switch to deno as runtime at the moment.

thank you.

**Reply by elycheikhsmail** [reply]
> Well Capacitor doesn't run on Node (except on Electron I suppose), it uses the web engine available on each platform. For example, that's the WebKit engine in `WKWebView` and the Chrome engine on Android. Neither of those are Node contexts, so the idea of swapping it out with a different JS engine, while possible for some non-web operations (such as those that would be done with `JavaScriptCore`), isn't possible for the rest of the app.

I understand this thank you.
